lib/git_hooks/installer.rb

```diff
@@ -1,33 +1,42 @@
 module GitHooks
   class Installer
-    attr_accessor :hook
-
     HOOK_SAMPLE_FILE = 'hook.sample'
 
-    def initialize(hook)
+    def initialize(hook, ruby_path = nil)
       @hook = hook
+      @ruby_path = ruby_path
     end
 
     def install(force = false)
-      return true if installed?
+      throw Exceptions::UnknownHookPresent.new(hook) if !force && installed?
+
+      hook_script = hook_template
```

Comment:
The first implementation used the `pre-commit` script directly into the root of the project. That script was then linked in `.git/hooks/`. I'm also not sure what the rationale behind the symbolic link was, so I came here to ask before chaning haha.

(Other problem with the symbolic link approach is that the hook is shared between all projects, since every hook installation would actually point to the `hook.sample` in the gem's directory. That doesn't seem wise in my opinion)

About the `installed?` validation: Yeah, I agree, this is pretty bad. We could just check the file existence, but this wouldn't guarantee the hook work.

One issue we have is that no check is performed to verify if the script has execution permissions (see my chmod 0750 somewhere).